api/src/main/java/jakarta/servlet/ServletResponse.java

```diff
@@ -182,17 +193,27 @@
      * Sets the content type of the response being sent to the client, if the response has not been committed yet. The given
      * content type may include a character encoding specification, for example, <code>text/html;charset=UTF-8</code>. The
      * response's character encoding is only set from the given content type if this method is called before
-     * <code>getWriter</code> is called.
+     * {@link #getWriter()} is called.
      * <p>
      * This method may be called repeatedly to change content type and character encoding. This method has no effect if
      * called after the response has been committed. It does not set the response's character encoding if it is called after
      * <code>getWriter</code> has been called or after the response has been committed.
      * <p>
+     * If calling this method has an effect (as per the previous paragraph), calling this method with {@code null} clears
+     * any content type set via a previous call to this method and clears any character encoding set via a previous call to
+     * this method, {@link #setContentType} or {@link #setLocale} but does not affect any default character encoding
+     * configured via {@link ServletContext#setResponseCharacterEncoding} or the deployment descriptor.
+     * <p>
+     * Calling this method with an invalid or unrecognised character encoding has no effect until {@link #getWriter()} is
+     * called at which point a {@link UnsupportedEncodingException} will be thrown.
```

Comment:
Not sure I like this.  Firstly, what if getWriter is never called and instead getOutputStream is used to write directly encoded bytes?
In that case is there an effect of the call with invalid or unrecognised character encoding?  Ie is the Content-Type header sent with the unknown charset?       If that is what we want, then we should say so explicitly here, otherwise I suggest we throw and IllegalArgumentException to be thrown in this method.

I think I lean towards allowing unknown encodings and only throwing if getWriter is called.  Perhaps it should say:
```suggestion
     * If this method is called with an unrecognised character encoding, then a subsequent call to {@link #getWriter()}
     * will throw an  {@link UnsupportedEncodingException} will be thrown.  Content for an unknown encoding can be sent 
     * with the <code>OutputStream</code> returned from {@link #getOutputStream}
```

However, perhaps we should throw `IllegalArgumentException` if the content type is badly formed (eg unclosed quotes or illegal characters).

Reply:
I like the new text. I'll update the PR.

We had a user of Tomcat that needed to set an invalid content-type header due to a broken client. With that in mind I'd prefer to leave the current behaviour as is.